packages/rest/lib/resource.cadl

```diff
@@ -110,3 +110,89 @@ interface ResourceInstanceOperations<TResource, TError> mixes ResourceRead<TReso
 interface ResourceCollectionOperations<TResource, TError> mixes ResourceCreate<TResource, TError>, ResourceList<TResource, TError> {}
 
 interface ResourceOperations<TResource, TError> mixes ResourceInstanceOperations<TResource, TError>, ResourceCollectionOperations<TResource, TError> {}
+
+interface SingletonResourceRead<TSingleton, TResource, TError> {
+  @autoRoute
+  @doc("Gets the singleton resource.")
+  @segmentOf(TSingleton)
+  @readsResource(TSingleton)
+  Get(...ResourceParameters<TResource>): TSingleton | TError;
+}
+
+interface SingletonResourceUpdate<TSingleton, TResource, TError> {
```

Comment:
I think singletons can also have create and delete in some cases - i.e. there is either one or none.  This often happens in child resources, where this can indicate presence or absence of some configured property. Not sure which pattern is more prevalent, though.

Reply:
I was looking for a case where a singleton has a `DELETE` and I found this weird one:

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/blob.json#L497-L720

This seems to be a singleton resource where you can give an arbitrary name for GET/PUT/DELETE but there are actions tied to a singleton resource for that type named `default`.  Ever seen that before?

Reply:
This looks like a grandchild of a singleton resource for blobServices.  In this case, the singleton might or might not exist as a consequence of how the storage account itself is created, but there is no facility inside the data plane for creating or deleting the singleton.

When I am thinking of createOrUpdate for a singleton, I am thinking of something more liek this: https://github.com/Azure/azure-rest-api-specs/blob/88e7838a09868a51de3894114355c75929847a46/specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-06-01-preview/workspace.json#L682-L779